examples/md-flexible/src/domainDecomposition/DomainTools.h

```diff
@@ -0,0 +1,65 @@
+ */
+#pragma once
+
+#include <array>
+#include <vector>
+
+namespace DomainTools {
+/**
+ * Checks if the provided coordinates are inside the domain defined by boxMin and boxMax.
+ * @param coordinates The coordinates in question.
+ * @param boxMin The minimum boundaries of the box domain.
+ * @param boxMax the maximum boundaries of the box domain.
+ * $return true if the coordinates lie within the provided box.
+ */
+bool isInsideDomain(const std::vector<double> &coordinates, std::vector<double> &boxMin, std::vector<double> &boxMax);
+
+/**
+ * Checks if the provided coordinates are inside the domain defined by boxMin and boxMax.
+ * Instead of a vector, the parameters are of type std::array<double, 3> to be compatible with AutoPas.
+ * @param coordinates The coordinates in question.
+ * @param boxMin The minimum boundaries of the box domain.
+ * @param boxMax the maximum boundaries of the box domain.
+ * $return true if the coordinates lie within the provided box.
+ */
+bool isInsideDomain(const std::array<double, 3> &coordinates, std::array<double, 3> &boxMin,
+                    std::array<double, 3> &boxMax);
+
+/**
+ * Calculates the distance of provided coordinates to the domain defined by boxMin and boxMax.
+ * If the coordinates lie inside the domain, the function returns 0.
+ * If the coordinates and the domain do not have the same number of dimensions, the function returns -1.
+ * @param coordinates The coordinates in question.
+ * @param boxMin The minimum boundaries of the box domain.
+ * @param boxMax the maximum boundaries of the box domain.
+ * @return the distance of the coordinates to the provided  domain.
+ */
+double getDistanceToDomain(const std::vector<double> &coordinates, std::vector<double> &boxMin,
+                           std::vector<double> &boxMax);
+
+/**
+ * Calculates the distance of provided coordinates to the domain defined by boxMin and boxMax.
+ * Instead of a vector, the parameters are of type std::array<double, 3> to be compatible with AutoPas.
+ * If the coordinates lie inside the domain, the function returns 0.
+ * If the coordinates and the domain do not have the same number of dimensions, the function returns -1.
+ * @param coordinates The coordinates in question.
+ * @param boxMin The minimum boundaries of the box domain.
+ * @param boxMax the maximum boundaries of the box domain.
+ * @return the distance of the coordinates to the provided  domain.
+ */
+double getDistanceToDomain(const std::array<double, 3> &coordinates, std::array<double, 3> &boxMin,
+                           std::array<double, 3> &boxMax);
+
+/**
+ * Generates a decomposition with a secific number of subdomains.
+ * This function uses prime factorization to determine the number of subdomains in each dimension.
+ * @param subdomainCount The number of subdomains in the resulting decomposition
+ * @param dimensionCount The number of dimensions in the simulation.
+ * @param oDecomposition The resulting decomposition.
```

Comment:
```suggestion
 * @param oDecomposition Vector containing the number of subdomains per dimension.
```
is this interpretation correct?
Should probably be an array?
And why "o" ?

Reply:
"o" is a convention to mark "output" parameters, i. e. funciton arguments which will be changed by a function. We use this convention where I work. I will remove it.
The interpretation is rather vague, I agree. This vector only contains the number of subdomains along each dimension.
I chose the type vector with an arbitrary amount of dimensions in mind.